### PR TITLE
compute sessions

### PR DESCRIPTION
**Pull request by mana-alisafaee**

### CODE REVIEW

File: components/renku_data_services/session/api.spec.yaml [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+        creation_date:
+          $ref: "#/components/schemas/CreationDate"
+        description:
+          $ref: "#/components/schemas/Description"
+        environment_id:
+          $ref: "#/components/schemas/EnvironmentId"
+        status:
+          $ref: "#/components/schemas/Status"
+        tags:
+          $ref: "#/components/schemas/TagsList"
```
Comment (leafty) [inline components/renku_data_services/session/api.spec.yaml:142]: I don't think we have anything yet in the pitch about tags. We would have to understand where to put them between sessions and environments, so I'm not sure we can use them now.

File: components/renku_data_services/session/api.spec.yaml [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+        - sessions
+components:
+  schemas:
+    SessionsList:
+      description: A list of Renku sessions
+      type: array
+      items:
+        $ref: "#/components/schemas/Session"
+      minItems: 0
+    Session:
```
Comment (leafty) [inline components/renku_data_services/session/api.spec.yaml:124]: A session belongs to a project. I think there is a missing `project_id`.

File: components/renku_data_services/session/api.spec.yaml (hunk 1 above)
Comment (mana-alisafaee) [reply components/renku_data_services/session/api.spec.yaml:142]: I'll remove them for now.

File: components/renku_data_services/session/api.spec.yaml [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+info:
+  title: Renku 1.0 Environment and Session Management Service
+  description: |
+    Service that allows creating, deleting, and managing Renku 1.0 environments and sessions.
+  version: 0.1.0
+servers:
+  - url: /api/data
+  - url: /ui-server/api/data
+paths:
+  /sessions:
```
[the remark this answers is not in the document]
Comment (mana-alisafaee) [reply components/renku_data_services/session/api.spec.yaml:11]: As discussed, this will be implemented later in this pitch.

**Review by leafty — COMMENTED** [review]
You need to add `run_migrations_for_app("sessions")` in `bases/renku_data_services/data_api/main.py` so that the new table is created.

**Comment by leafty** [comment]
Should we merge this into a feature branch? Or do we consider this the feature branch?

File: components/renku_data_services/session/blueprints.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    authenticator: base_models.Authenticator
+
+    def get_all(self) -> BlueprintFactoryResponse:
+        """List all sessions."""
+
+        @authenticate(self.authenticator)
+        async def _get_all(_: Request, *, user: base_models.APIUser):
+            sessions = await self.session_repo.get_sessions(user=user)
+            return json(
+                [apispec.Session.model_validate(p).model_dump(exclude_none=True, mode="json") for p in sessions], 201
```
Comment (leafty) [inline components/renku_data_services/session/blueprints.py:33]: This should send back a 200:

```suggestion
                [apispec.Session.model_validate(p).model_dump(exclude_none=True, mode="json") for p in sessions]
```

**Comment by mana-alisafaee** [comment]
> Should we merge this into a feature branch? Or do we consider this the feature branch?

We can consider this as the feature branch.

**Pull request merged**